Buildings/Applications/DHC/EnergyTransferStations/Combined/Generation5/Controls/SideCold.mo

```diff
@@ -67,19 +61,26 @@ model SideCold "Control block for cold side"
     annotation (Placement(transformation(extent={{-70,-170},{-50,-150}})));
   Buildings.Controls.OBC.CDL.Continuous.Max max1 "CHWST reset signal"
     annotation (Placement(transformation(extent={{-90,-90},{-70,-70}})));
-  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant zer1(k=0)
-                                                                  "Zero"
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant zer1(k=0) "Zero"
     annotation (Placement(transformation(extent={{-150,-70},{-130,-50}})));
   Buildings.Controls.OBC.CDL.Continuous.AddParameter addPar1(p=-nSouAmb, k=1)
     "Control signal minus nSouAmb"
     annotation (Placement(transformation(extent={{-150,-110},{-130,-90}})));
   Buildings.Controls.OBC.CDL.Continuous.Min min1
     "Ambient source control signal"
     annotation (Placement(transformation(extent={{-90,-130},{-70,-110}})));
+  Buildings.Controls.OBC.CDL.Continuous.MultiMax mulMax(
+    final nin=nSouAmb)
+    "Maximum value"
+    annotation (Placement(transformation(extent={{90,-130},{110,-110}})));
```

Comment:
This new block is not needed. The maximum of the elements of the output array from `mapFun[nSouAmb]` will always be at index 1. Hence the previous connect clause `connect(mapFun[1].y, greThr.u)`. Did you make the change because you thought it was incorrect or because you thought it was clearer for the user? In the latter case I will go with your modification.